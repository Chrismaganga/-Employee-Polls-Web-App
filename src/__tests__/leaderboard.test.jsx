/* eslint-disable no-undef */
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Leaderboard from '../components/Leaderboard'

const mockStore = (state) => {
  return createStore((state) => state, state)
}

describe('Leaderboard Component', () => {
  it('should match snapshot', () => {
    const mockUsers = {
      'user1': {
        id: 'user1',
        name: 'John Doe',
        avatarURL: "https://ui-avatars.com/api/?background=0D8ABC&color=fff&name=John+Doe",
        questions: ['q1', 'q2'],
        answers: { 'q3': 'optionOne' }
      },
      'user2': {
        id: 'user2',
        name: 'Jane Smith',
        avatarURL: "https://ui-avatars.com/api/?background=0D8ABC&color=fff&name=Jane+Doe",
        questions: ['q4'],
        answers: { 'q3': 'optionTwo' }
      }
    }

    const store = mockStore({
      users: mockUsers
    })

    const { asFragment } = render(
      <Provider store={store}>
        <Leaderboard />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
