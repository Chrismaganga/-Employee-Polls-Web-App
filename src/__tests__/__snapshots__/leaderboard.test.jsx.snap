// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="leaderboard"
  >
    <h1
      class="leaderboard-heading"
    >
       Leaderboard 
    </h1>
    <ul>
      <li>
        <div
          class="board-entry"
        >
          <div
            class="board-left"
          >
            <img
              alt="john doe avatar"
              height="40"
              src="https://ui-avatars.com/api/?background=0D8ABC&color=fff&name=John+Doe"
              width="40"
            />
            <p>
              John Doe
            </p>
          </div>
          <div
            class="board-right"
          >
            <p>
              Questions: 2
            </p>
            <p>
              Answers: 1
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          class="board-entry"
        >
          <div
            class="board-left"
          >
            <img
              alt="jane smith avatar"
              height="40"
              src="https://ui-avatars.com/api/?background=0D8ABC&color=fff&name=Jane+Doe"
              width="40"
            />
            <p>
              Jane Smith
            </p>
          </div>
          <div
            class="board-right"
          >
            <p>
              Questions: 1
            </p>
            <p>
              Answers: 1
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
